modules/gatk4/variantrecalibrator/main.nf

```diff
@@ -0,0 +1,70 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process GATK4_VARIANTRECALIBRATOR {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+        conda (params.enable_conda ? "bioconda::gatk4=4.2.3.0" : null)
+        if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+            container "https://depot.galaxyproject.org/singularity/gatk4:4.2.3.0--hdfd78af_0"
+        } else {
+            container "quay.io/biocontainers/gatk4:4.2.3.0--hdfd78af_0"
+        }
+
+    input:
+    tuple val(meta), path(vcf) , path(tbi)
+
+    path fasta
+    path fai
+    path dict
+    val allelespecific
+    path resource_vcfs
+    path resource_tbis
+    val resource_labels
```

Comment:
These are not independent inputs: wouldn't be better to pass these in a structured way? 
[ (label, vcf, tbi),
  (label, vcf, tbi)
]
you could then build your "resourceCommand" from it, and catch two birds with one stone, i.e. having both the labels and the input files associated to them?
Maybe I'm missing something here?

Reply:
might be able to do it: 
[ (label, label, label), (vcf, vcf, vcf), (tbi, tbi, tbi) ] 
as then they could be specified on input as : 
tuple val(label), path(vcf) , path(tbi)
and it would work regardless of how many resources are specified.

unsure how to make [ (label, vcf, tbi), (label, vcf, tbi), (label, vcf, tbi) ] work as then you have a variable sized tuple of tuples going into the module. With the tuples consisting of a mix of vals and paths, so you couldn't just declare the entire tuple as a val input.